public/editor/scripts/editor/js/fc/publisher.js

```diff
@@ -173,52 +173,54 @@ define(function(require) {
     var dialog = publisher.dialog;
     var buttons = dialog.buttons;
 
-    if (publisher.unpublishing) {
-      return;
-    }
-    publisher.unpublishing = true;
-
-    function setState(done) {
-      buttons.publish[done ? "on" : "off"]("click", handlers.publish);
-      buttons.unpublish.children("span").text(done ? TEXT_UNPUBLISH : TEXT_UNPUBLISHING);
-    }
-
-    // Disable all actions during the unpublish
-    buttons.unpublish.off("click", handlers.unpublish);
-    setState(false);
-    SyncState.syncing();
-
-    var request = publisher.generateRequest("unpublish");
-    request.done(function() {
-      if(request.status !== 200) {
-        console.error("[Thimble] Server was unable to unpublish project, responded with status ", request.status);
-        buttons.unpublish.on("click", handlers.unpublish);
+    if (window.confirm("Are you sure you want to unpublish this project? Links that have already been shared will no longer work.")) {
+      if (publisher.unpublishing) {
         return;
       }
+      publisher.unpublishing = true;
 
-      buttons.parent.removeClass("hide");
+      var setState = function(done) {
```

Comment:
agreed, only added it to avoid a declaration inside an if statement, which I've moved as per your above comments 👍